src/main/java/org/skriptlang/skript/elements/fireworks/ExprAttachedTo.java

```diff
@@ -0,0 +1,99 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.elements.fireworks;
+
+import org.bukkit.entity.Firework;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Firework Attached To")
+@Description("Returns the entity attached to the fireworks if any.")
+@Examples({
+	"on firework explode:",
+		"\tloop all players in radius 10 around event-location:",
+			"\t\tloop-player is not attached entity of event-firework",
+			"\t\tpush loop-player upwards at speed 2"
+})
+@RequiredPlugins("Spigot 1.19.4+ or Paper")
+@Since("INSERT VERSION")
+@SuppressWarnings("deprecation")
+public class ExprAttachedTo extends SimplePropertyExpression<Firework, LivingEntity> {
+
+	/**
+	 * Developer note:
+	 * Paper had a method called getBoostedEntity() in the Firework class since before 1.13.
+	 * Spigot added the same method but called it getAttachedTo() in 1.19.
+	 */
+	private static final boolean BOOSTED_ENTITY = Skript.methodExists(Firework.class, "getBoostedEntity");
+	private static final boolean RUNNING_1_19 = Skript.isRunningMinecraft(1, 19, 4);
+
+	static {
+		if (RUNNING_1_19 || BOOSTED_ENTITY)
+			register(ExprAttachedTo.class, LivingEntity.class, "(attached [to]|boost(ing|ed)) entity", "fireworks");
+	}
```

Comment:
Could this be a `registerDefault()` expression? Or do we think current syntax might be misleading without the firework entity

Reply:
It works for boosted entity, but not really the other, which is why it wasn't. Added

Reply:
Yeah that's about what I thought, seemed like attached didn't